### DISCUSSION TITLE
Add granular control to defer

### DISCUSSION THREAD

**Discussion by jinglesthula** [remix-run/remix#5265] [Proposals]
I propose, adding a couple things to augment defer.

It would be great to indicate that you want to defer for initial load, but not for revalidation.

The corollary is to also add support the opposite where you can indicate that you want to wait on initial page load, but defer for revalidation.

I’m not sure whether it makes sense to Allow for further differentiation between server navigation and client navigation. No real use case where you might want to be able to do one or the other of those comes to mind immediately, but maybe some people can think of one more that would be really handy.

**Comment by alfl** [comment]
Would love to be able to defer loads on the server. Related: https://github.com/remix-run/remix/discussions/5226
